### PR TITLE
Typescript definitions

### PR DESCRIPTION
**Pull request by phobon**
So, I've been using styled system with typescript and styled-components with this definition file. It was sourced originally from https://github.com/jxnblk/styled-system/issues/169 and extended to support all of the different functions.

I don't think it's perfect, but it's been working really well for me - just interested to see what everyone thinks. This is my first time contributing to something, I just love this project 👍

### CODE REVIEW

**Comment by msokk** [comment]
Looks like styled-components will be moving definitions out of the repo to DefinitelyTyped in v4, maybe we should add these to DT too instead?

**Comment by phobon** [comment]
I don't think that's a bad idea at all, probably makes sense to maintain things there as well

File: src/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,460 @@
...
+   * Utilities
+   */
+  export function theme(keys: string): any;
+  export function cleanElement(component: any): any;
+  export function removeProps(props: any): any;
+
+  /**
+   * Low level style functions
+   */
+  export interface LowLevelStyleFunctionArguments {
```
Comment (smashercosmo) [inline src/index.d.ts]: 'alias' property is missing

**Comment by jxnblk** [comment]
Thanks, but as I've discussed in many, many other issues, is there any reason to not host this in [DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped)?

I don't use TS and won't be likely to keep this maintained over time

**Comment by phobon** [comment]
That's all good - no reason at all why this can't live in DT. I'll make a PR there 👍

**Comment by zephraph** [comment]
@phobon I've started a PR @ https://github.com/DefinitelyTyped/DefinitelyTyped/pull/26649. Do you want to be a maintainer on it? I've got you listed as an author currently.

**Comment by phobon** [comment]
@zephraph Sorry about the late reply - thanks for setting that all up - looks really good! More than happy to be a maintainer 👍

**Pull request closed without merging**